src/Checkbox/Checkbox.jsx

```diff
@@ -27,18 +28,20 @@ export const Checkbox = ({
 
   const handleChange = useCallback(
     event => {
-      const { checked: nativeChecked } = event.target
-      setChecked(nativeChecked)
-      onChange && onChange(event)
+      if (!disabled) {
+        const { checked: nativeChecked } = event.target
+        setChecked(nativeChecked)
+        onChange && onChange(event)
+      }
     },
-    [setChecked, onChange]
+    [setChecked, onChange, disabled]
```

Comment:
This circumvents the built-in behaviour. The `disabled` attribute is missing from the native `input type="checkbox"` and by adding it, it will fix the bug, because the input will not change if disabled and the onChange event is not called.

As additional guard, the `StyledLabel` should have `pointer-events: none` when disabled, so a click or tap will not trigger an event.